File: JobSearchCodeExamples/JobSearchCodeExamples/TestTaker/EmployeeCoverage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JobSearchCodeExamples.TestTaker;

/// <summary>
/// Routines to calculate employee coverage
/// </summary>
public static class EmployeeCoverage
{
    /// <summary>
    /// Maximums the coverage.
    /// </summary>
    /// <param name="employees">The employees.</param>
    /// <returns></returns>
    public static int MaxCoverage(string[] employees)
    {
        int[] combined = new int[employees.Length];

        for (int i = 0; i < employees.Length; i++)
        {
            string[] years = employees[i].Split("-");
            int startDate = int.Parse(years[0]);
            int endDate = int.Parse(years[1]);
            for (int j = i + 1; j < employees.Length; j++)
            {
                string[] years2 = employees[j].Split("-");
                int start = int.Parse(years2[0]);
                int end = int.Parse(years2[1]);
                if (startDate <= start && endDate >= start ||
                    startDate <= end && endDate >= end ||
                    startDate <= start && endDate >= end ||
                    startDate >= start && endDate <= end)
                {
                    combined[i] += combined[i] == 0 ? 2 : 1;
                    combined[j] += combined[j] == 0 ? 2 : 1;
                }
            }
        }
        var maxValue = 0;
        for (int i = 0; i < employees.Length; i++)
            if (combined[i] > maxValue) maxValue = combined[i];

        return maxValue;
    }
}

File: JobSearchCodeExamples/JobSearchCodeExamples.FSharp/TestTaker/EmployeeCoverage.fs
namespace JobSearchCodeExamples.FSharp.TestTaker

module EmployeeCoverage =
    let MaxCoverage (employees : string[]) : int =
        let combined : int array = Array.zeroCreate (employees.Length)
        for i = 0 to employees.Length - 1 do
            let years = employees[i].Split("-")
            let startDate = years[0] |> int
            let endDate = years[1] |> int
            for j = i + 1 to employees.Length - 1 do
                let years2 = employees[j].Split("-")
                let start = years2[0] |> int
                let endYear = years2[1] |> int
                if startDate <= start && endDate >= start ||
                   startDate <= endYear && endDate >= endYear ||
                   startDate <= start && endDate >= endYear ||
                   startDate >= start && endDate <= endYear then
                   combined[i] <- combined[i] + if combined[i] = 0 then 2 else 1
                   combined[j] <- combined[j] + if combined[j] = 0 then 2 else 1
        let mutable maxValue = 0
        for i = 0 to employees.Length - 1 do
            if combined[i] > maxValue then maxValue <- combined[i]
        maxValue
                    

